packages/async-flow/src/types.d.ts

```diff
@@ -60,6 +67,8 @@ export type HostOf<F> = F extends (...args: infer A) => Promise<infer R>
   ? (...args: A) => Vow<R extends Passable ? R : HostInterface<R>>
   : F;
 
+export type HostArgs<GA extends any[]> = { [K in keyof GA]: HostOf<GA[K]> };
```

Comment:
I'm surprised this works, I would have expected some kind of tuple here, but I'm rusty on my TS mapped types

Reply:
object properties are how TS maps tuples